xcm/xcm-builder/src/origin_aliases.rs

```diff
@@ -15,65 +15,32 @@
 // along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
 
 use frame_support::traits::{ContainsPair, Get};
+use sp_std::marker::PhantomData;
 use xcm::latest::prelude::*;
 
-pub struct AliasSiblingAccountId32<ParachainId>(sp_std::marker::PhantomData<ParachainId>);
-impl<ParachainId: Get<u32>> ContainsPair<MultiLocation, MultiLocation>
-	for AliasSiblingAccountId32<ParachainId>
+pub struct RemovePrefixAccountId32<Prefix>(PhantomData<Prefix>);
+impl<Prefix: Get<MultiLocation>> ContainsPair<MultiLocation, MultiLocation>
+	for RemovePrefixAccountId32<Prefix>
 {
 	fn contains(origin: &MultiLocation, target: &MultiLocation) -> bool {
-		match origin {
-			MultiLocation {
-				parents: 1,
-				interior:
-					X2(Parachain(para_id), origin_account  @  AccountId32 {network: _, id: _, }),
-			} if *para_id == ParachainId::get() => match target {
-				MultiLocation {
-					parents: 0,
-					interior: X1(alias_account),
-				} if origin_account == alias_account => true,
-				_ => false,
-			},
-			_ => false,
+		if let Ok(appended) = (*target).prepended_with(Prefix::get()) {
```

Comment:
This doesn't do what you think it does -- it prepends `target` with `Prefix`, which is not what you want. You want to _check_ whether `target` contains `Prefix` as the prefix.

Reply:
In other words, you're looking for `starts_with`.